controllers/volumereplication_controller.go

```diff
@@ -162,28 +171,42 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	case replicationv1alpha1.Primary:
 		err := r.markVolumeAsPrimary(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Secondary:
 		err := r.markVolumeAsSecondary(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Resync:
 		err := r.resyncVolume(volumeHandle, parameters, secret)
 		if err != nil {
+			r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	default:
+		r.updateStatus(instance, replicationv1alpha1.ReplicationFailure, ("unsupported image state " + string(instance.Spec.ImageState)))
 		return ctrl.Result{}, fmt.Errorf("unsupported image state %q", instance.Spec.ImageState)
 	}
 
+	instance.Status.LastCompletionTime = metav1.Now()
+	r.updateStatus(instance, replicationv1alpha1.Replicating, "volume replication is ready to use")
 	return ctrl.Result{}, nil
 }
 
+func (r *VolumeReplicationReconciler) updateStatus(instance *replicationv1alpha1.VolumeReplication, state replicationv1alpha1.ReplicationState, message string) {
+	instance.Status.State = state
+	instance.Status.Message = message
+	if err := r.Client.Update(context.TODO(), instance); err != nil {
+		r.Log.Error(err, "Update Error", "MetaUpdateErr", "Failed to update status")
```

Comment:
The object status is metadata right that's why I wrote like this. even in the ocs-operator we do the same, otherwise, we can change this to the `ObjectUpdateErr` or something like that.

Reply:
can this be just `r.Log.Error(err, "Update Error", "Failed to update status")`

Reply:
Updated in diff way